### PR TITLE
Remove dead codes

### PR DESCRIPTION
**Pull request by sidepelican**
There is a lot of dead code in mockolo due to historical reasons.
With many unused definitions and arguments, they make hard to understand of the code, so I would like to clean up them.

In this PR, I will make some simple deletions and replace some implementations with simpler forms

### CODE REVIEW

File: Sources/MockoloFramework/Parsers/SwiftSyntaxExtensions.swift [hunk 1]
```diff
@@ -30,13 +39,6 @@ extension SyntaxProtocol {
     }
 }
 
-
-extension AttributeListSyntax {
-    var trimmedDescription: String? {
```
Comment (sidepelican) [inline Sources/MockoloFramework/Parsers/SwiftSyntaxExtensions.swift:35]: SwiftSyntax already has `trimmedDescription`

File: Sources/MockoloFramework/Templates/NominalTemplate.swift [hunk 2]
```diff
@@ -270,21 +270,14 @@ extension NominalModel {
...
-                }
+        var aliasMap = [String: Set<String>]()
+        for (_, model) in models {
+            if let alias = model as? TypeAliasModel {
+                aliasMap[alias.name, default: []].insert(alias.type.typeName)
             }
         }
         let aliasDupes = aliasMap.filter {$0.value.count > 1}
-        return aliasDupes.isEmpty ? nil : aliasDupes
+        return aliasDupes.isEmpty ? nil : aliasDupes.mapValues {$0.sorted()}
```
Comment (sidepelican) [inline Sources/MockoloFramework/Templates/NominalTemplate.swift:280]: The results are now sorted, but I believe there are no compatibility issues.

File: Sources/MockoloFramework/Models/ParamModel.swift [hunk 3]
```diff
@@ -17,19 +17,26 @@
 import Foundation
 
 final class ParamModel: Model {
-    var name: String
-    var offset: Int64
-    var length: Int64
-    var type: SwiftType
+    internal init(label: String, name: String, type: SwiftType, isGeneric: Bool, inInit: Bool, needsVarDecl: Bool, offset: Int64, length: Int64) {
```
Comment (uhooi) [inline Sources/MockoloFramework/Models/ParamModel.swift:20]: [Question]
Why is it explicitly marked internal?

File: Sources/MockoloFramework/Operations/UniqueModelGenerator.swift [hunk 4]
```diff
@@ -52,13 +51,15 @@ private func generateUniqueModels(key: String,
...
+    let (nonMethodModels, methodModels) = models.partitioned(by: { $0.modelType == .method })
     let orderedModels = [nonMethodModels, methodModels].flatMap {$0}
-    let x = uniqueEntities(in: orderedModels, exclude: processedLookup, fullnames: processedFullNames)
-    let unmockedUniqueEntities = x.filter {!$0.value.processed}
-    
+    let unmockedUniqueEntities = uniqueEntities(
+        in: orderedModels,
+        exclude: processedLookup,
+        fullnames: processedFullNames
+    ).filter {!$0.value.processed}
```
Comment (uhooi) [inline Sources/MockoloFramework/Operations/UniqueModelGenerator.swift:61]: [nits]
I need a space before and after the brace.

```suggestion
    ).filter { !$0.value.processed }
```

**Review by uhooi — APPROVED** [review]

File: Sources/MockoloFramework/Models/ParamModel.swift (hunk 3 above)
Comment (sidepelican) [reply Sources/MockoloFramework/Models/ParamModel.swift:20]: It doesn't have any particular meaning. It gets added when I used 'Generate Memberwise Initializer'. I left it as is.

File: Sources/MockoloFramework/Operations/UniqueModelGenerator.swift (hunk 4 above)
Comment (sidepelican) [reply Sources/MockoloFramework/Operations/UniqueModelGenerator.swift:61]: In mockolo's style, it seems that spaces are not used here.

**Pull request merged**